### PR TITLE
CLN: Simplify to_masked implementation

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by datapythonista — APPROVED** [review]
Very nice clean up, thanks @phofl

File: pandas/core/arrays/arrow/array.py [hunk 1]
```diff
@@ -1729,42 +1730,18 @@ def _replace_with_mask(
...
+            na_value = 1
         elif pa.types.is_boolean(pa_dtype):
-            dtype = "boolean"
-            np_dtype = "bool"
             na_value = True
-            arr_cls = BooleanArray
         else:
             raise NotImplementedError
 
+        dtype = _arrow_dtype_mapping()[pa_dtype]
```
Comment (jbrockmendel) [inline pandas/core/arrays/arrow/array.py:1741]: could we simplify even more by just doing `self.dtype.numpy_dtype`?  Aside from that, LGTM

Comment (phofl) [reply pandas/core/arrays/arrow/array.py:1741]: I'd prefer keeping the mask stuff, this should make the conversion faster

Comment (jbrockmendel) [reply pandas/core/arrays/arrow/array.py:1741]: Sounds good.

**Review by jbrockmendel — APPROVED** [review]

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**